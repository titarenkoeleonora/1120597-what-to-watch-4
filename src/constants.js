const emptyMovie = {
  id: 0,
  title: `Is loading...`,
  poster: ``,
  previewImage: ``,
  bgImage: ``,
  backgroundColor: ``,
  src: ``,
  previewVideoLink: ``,
  description: ``,
  rating: ``,
  votes: 0,
  director: ``,
  starring: [],
  runTime: ``,
  genre: ``,
  date: 0,
  isFavorite: ``,
};

const MovieTabs = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

const MAX_SHOWN_MOVIES = 8;

const API_TIMEOUT = 5000;

export {emptyMovie, MovieTabs, MAX_SHOWN_MOVIES, API_TIMEOUT};
