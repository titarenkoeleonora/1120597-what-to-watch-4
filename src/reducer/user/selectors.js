import NameSpace from "../name-space.js";

export const getAuthorizationStatus = (state) => state[NameSpace.USER].authorizationStatus;

export const getAuthorizationInfo = (state) => state[NameSpace.USER].authorizationInfo;

export const getIsAuthorizing = (state) => state[NameSpace.USER].isAuthorizing;
