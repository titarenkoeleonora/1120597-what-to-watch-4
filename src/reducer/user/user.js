import {UserActionCreator} from "../actions/user-action-creator";
import {UserActionType} from "../actions/user-action-type";
import {extend} from "../../utils";
import createUserInfo from "../../adapter/user";

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  isAuthorizing: false,
  authorizationInfo: {
    id: 0,
    email: ``,
    name: ``,
    avatar: ``,
  }};

const Operation = {
  checkAuth: () => (dispatch, getState, api) => {
    return api.get(`/login`)
      .then((response) => {
        dispatch(UserActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
        dispatch(UserActionCreator.loadAuthorizationInfo(createUserInfo(response.data)));
      })
      .catch((err) => {
        throw err;
      });
  },

  login: (authData) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.login,
      password: authData.password,
    })
      .then((response) => {
        dispatch(UserActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
        dispatch(UserActionCreator.loadAuthorizationInfo(createUserInfo(response.data)));
        dispatch(UserActionCreator.isNotAuthorizing());
      })
      .catch(() => {
        dispatch(UserActionCreator.requireAuthorization(AuthorizationStatus.NO_AUTH));
      });
  },
};


const reducer = (state = initialState, action) => {
  switch (action.type) {
    case UserActionType.REQUIRED_AUTHORIZATION:
      return extend(state, {
        authorizationStatus: action.payload,
      });
    case UserActionType.AUTHORIZATION_INFO:
      return extend(state, {
        authorizationInfo: action.payload
      });
    case UserActionType.IS_AUTHORIZING:
      return extend(state, {
        isAuthorizing: true,
      });
    case UserActionType.IS_NOT_AUTHORIZING:
      return extend(state, {
        isAuthorizing: false,
      });
  }

  return state;
};

export {reducer, initialState, AuthorizationStatus, Operation};
