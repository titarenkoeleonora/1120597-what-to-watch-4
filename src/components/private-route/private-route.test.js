import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from "redux-mock-store";
import PrivateRoute from './private-route.jsx';
import {AuthorizationStatus} from '../../reducer/user/user.js';
import {MemoryRouter} from 'react-router';
import {AppRoute} from '../../constants.js';
import NameSpace from '../../reducer/name-space.js';
import {Provider} from 'react-redux';

const MockComponent = () => <div />;
const mockStore = configureStore([]);

describe(`PrivateRouteSnapshot`, () => {
  it(`should render Loader`, () => {
    const store = mockStore({
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.NO_AUTH,
      }
    });

    const tree = renderer.create(
        <Provider store={store}>
          <MemoryRouter initialEntries={[AppRoute.MY_LIST]}>
            <PrivateRoute
              exact
              path={AppRoute.MY_LIST}
              authorizationStatus={AuthorizationStatus.AUTH}
              render={() => {
                return <MockComponent />;
              }}
            />
          </MemoryRouter>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should render mock component`, () => {
    const store = mockStore({
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.NO_AUTH,
      }
    });

    const tree = renderer.create(
        <Provider store={store}>
          <MemoryRouter initialEntries={[AppRoute.MY_LIST]}>
            <PrivateRoute
              exact
              path={AppRoute.MY_LIST}
              authorizationStatus={AuthorizationStatus.AUTH}
              render={() => {
                return <MockComponent />;
              }}
            />
          </MemoryRouter>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
