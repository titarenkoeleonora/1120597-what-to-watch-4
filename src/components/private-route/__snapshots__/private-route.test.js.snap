// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRouteSnapshot should render Loader 1`] = `null`;

exports[`PrivateRouteSnapshot should render mock component 1`] = `null`;
